update/Pos/Update/Poll/Logic/Base.hs

```diff
@@ -200,31 +200,29 @@ updateSlottingData
     :: (MonadError PollVerFailure m, MonadPoll m)
     => EpochIndex -> m ()
 updateSlottingData epoch = do
-    sd@SlottingData {..} <- getSlottingData
     let errFmt =
             ("can't update slotting data, stored penult epoch is "%int%
              ", while given epoch is "%int%
              ")")
-    if | sdPenultEpoch + 1 == epoch -> updateSlottingDataDo sd
+    let err = throwError $ PollInternalError $ sformat errFmt (-1) epoch
+    mres <- do
+      li <- getEpochLastIndex
+      fmap (li,) <$> getEpochSlottingData li
+    (lastIndex, esd) <- maybe err return mres
```

Comment:
Oh, sorry, `maybeThrow` is for `MonadThrow (throwM)`. But for this use case there is `note` in `Universum`, though I am not sure it's a good idea to use it. So I don't mind if you leave it as it is.